modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -0,0 +1,266 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Collection of algebra utility functions, used to shuffle data across partitions."""
+
+import numpy as np
+import pandas
+from typing import Callable, Union, Optional
+from collections import namedtuple
+
+from modin.core.dataframe.pandas.dataframe.dataframe import PandasDataframe
+
+
+SortFunctions = namedtuple(
+    "SortFunctions", ["sample_function", "pivot_function", "split_function"]
+)
+
+
+def build_sort_functions(
+    modin_frame: PandasDataframe,
+    column: str,
+    method: str,
+    ascending: Union[list, bool],
+    **kwargs: dict,
+) -> SortFunctions:
+    """
+    Return a dictionary containing the functions necessary to perform a sort.
+
+    Parameters
+    ----------
+    modin_frame : PandasDataframe
+        The frame calling these sort functions.
+    column : str
+        The major column name to sort by.
+    method : str
+        The method to use for picking quantiles.
+    ascending : bool
+        The ascending flag.
+    **kwargs : dict
+        Additional keyword arguments.
+
+    Returns
+    -------
+    dict :
+        A dictionary containing the functions to pick quantiles, pick overall quantiles, and split
+        partitions for sorting.
+    """
+
+    def sample_fn(partition):
+        return pick_samples_for_quantiles(
+            partition, column, len(modin_frame._partitions), len(modin_frame.index)
+        )
+
+    def pivot_fn(samples):
+        key = kwargs.get("key", None)
+        return pick_pivots_from_samples_for_sort(modin_frame, samples, method, key)
+
+    def split_fn(partition, pivots):
+        return split_partitions_using_pivots_for_sort(
+            modin_frame, partition, column, pivots, ascending, **kwargs
+        )
+
+    return SortFunctions(
+        sample_function=sample_fn, pivot_function=pivot_fn, split_function=split_fn
+    )
+
+
+def _find_quantiles(
```

Comment:
so this function could be removed after we merge https://github.com/modin-project/modin/pull/5229 ?

Reply:
No - this is because we keep `numpy >= 1.18` as a dependency - once we upgrade that dependency to a newer numpy version, then this function can be deleted. I checked #5229 and it doesn't seem to upgrade our base numpy dep.

Reply:
[numpy 1.18](https://pypi.org/project/numpy/1.18.0/) does still seem to support Python 3.8, so I think we need to support it for now.